src/python/pants/backend/python/lint/flake8/subsystem.py

```diff
@@ -117,6 +153,75 @@ def config_request(self) -> ConfigFilesRequest:
             check_content={"setup.cfg": b"[flake8]", "tox.ini": b"[flake8]"},
         )
 
+    @property
+    def source_plugins(self) -> UnparsedAddressInputs:
+        return UnparsedAddressInputs(self.options.source_plugins, owning_address=None)
+
+
+# --------------------------------------------------------------------------------------
+# First-party plugins
+# --------------------------------------------------------------------------------------
+
+
+@dataclass(frozen=True)
+class Flake8FirstPartyPlugins:
+    requirement_strings: FrozenOrderedSet[str]
+    interpreter_constraints_fields: FrozenOrderedSet[InterpreterConstraintsField]
+    sources_digest: Digest
+
+    PREFIX = "__plugins"
+
+    def __bool__(self) -> bool:
+        return self.sources_digest != EMPTY_DIGEST
+
+
+@rule("Prepare [flake8].source_plugins", level=LogLevel.DEBUG)
+async def flake8_first_party_plugins(flake8: Flake8) -> Flake8FirstPartyPlugins:
+    if not flake8.source_plugins:
+        return Flake8FirstPartyPlugins(FrozenOrderedSet(), FrozenOrderedSet(), EMPTY_DIGEST)
+
+    plugin_target_addresses = await Get(Addresses, UnparsedAddressInputs, flake8.source_plugins)
+    transitive_targets = await Get(
+        TransitiveTargets, TransitiveTargetsRequest(plugin_target_addresses)
+    )
+
+    requirements_fields: OrderedSet[PythonRequirementsField] = OrderedSet()
+    interpreter_constraints_fields: OrderedSet[InterpreterConstraintsField] = OrderedSet()
+    for tgt in transitive_targets.closure:
+        if tgt.has_field(PythonRequirementsField):
+            requirements_fields.add(tgt[PythonRequirementsField])
+        if tgt.has_field(InterpreterConstraintsField):
+            interpreter_constraints_fields.add(tgt[InterpreterConstraintsField])
+
+    # NB: Flake8 source plugins must be explicitly loaded via PYTHONPATH (i.e. PEX_EXTRA_SYS_PATH).
+    # The value must point to the plugin's directory, rather than to a parent's directory, because
+    # `flake8:local-plugins` values take a module name rather than a path to the module;
+    # i.e. `plugin`, but not `path/to/plugin`.
+    # (This means users must have specified the parent directory as a source root.)
```

Comment:
I'm not sure this is 100% true based on the docs, but I couldn't get it working in Pants with path/to/plugin.

I figured it's least complicated ot just make it match `pylint` :man_shrugging: